common/blockchain_util.py

```diff
@@ -108,6 +108,9 @@ def create_account(self):
         return account.address, account.privateKey.hex()
 
     def get_current_block_no(self):
+        logger.info(self.web3_object.isConnected())
+        if not self.web3_object.isConnected():
```

Comment:
Can isConnected() throw an exception? We should catch it and reset connection if an exception is thrown as well. Also please test this. Make sure that isCOnnected is indeed displaying the behaviour you expect

Reply:
This is how I tested locally
from web3 import Web3
w3 = Web3() 
w3.isConnected() # false
w3 = Web3(web3.providers.WebsocketProvider(URL)) 
w3.isConnected() # True
isConnected() throws websockets.exceptions.InvalidStatusCode Exception when URL is stale. We don't need to handle this exception.